src/blockchain_db/sqlite/db_sqlite.cpp

```diff
@@ -137,30 +161,38 @@ bool BlockchainSQLite::decrement_height()
 }
 
 
-bool BlockchainSQLite::add_sn_payments(std::vector<cryptonote::batch_sn_payment>& payments, uint64_t block_height)
+bool BlockchainSQLite::add_sn_payments(std::vector<cryptonote::batch_sn_payment>& payments)
 {
-  LOG_PRINT_L3("BlockchainDB_SQLITE::" << __func__ << " called on height: " << block_height);
-  //Assert that all the addresses in the payments vector are unique
-  std::sort(payments.begin(),payments.end(),[](const cryptonote::batch_sn_payment i, const cryptonote::batch_sn_payment j){ return i.address < j.address; });
-  auto uniq = std::unique( payments.begin(), payments.end(), [](const cryptonote::batch_sn_payment i, const cryptonote::batch_sn_payment j){ return i.address == j.address; } );
-  if (uniq != payments.end()) {
-    MWARNING("Duplicate addresses in payments list");
-    return false;
-  }
-
+  LOG_PRINT_L3("BlockchainDB_SQLITE::" << __func__);
   SQLite::Statement insert_payment{db,
-    "INSERT INTO batched_payments (address, amount, height_earned) VALUES (?, ?, ?)"};
+    "INSERT INTO batched_payments_accrued (address, amount) VALUES (?1, ?2) ON CONFLICT (address) DO UPDATE SET amount = amount + (?2)"};
 
   for (auto& payment: payments) {
     std::string address_str = cryptonote::get_account_address_as_str(m_nettype, 0, payment.address_info.address);
     MTRACE("Adding record for SN reward contributor " << address_str << "to database with amount " << static_cast<int64_t>(payment.amount));
-    db::exec_query(insert_payment, address_str, static_cast<int64_t>(payment.amount), static_cast<int64_t>(block_height));
+
+    db::exec_query(insert_payment, address_str, static_cast<int64_t>(payment.amount));
     insert_payment.reset();
   };
 
   return true;
 }
 
+bool BlockchainSQLite::subtract_sn_payments(std::vector<cryptonote::batch_sn_payment>& payments)
+{
+  LOG_PRINT_L3("BlockchainDB_SQLITE::" << __func__);
+	SQLite::Statement update_payment{db,
+		"UPDATE batched_payments_accrued SET amount = (amount - ?) WHERE address = ?"};
```

Comment:
This query can fail to update any rows, which could happen if the last update to accrued set amount to 0 (which then has a trigger to remove the row).  This would indicate a bug elsewhere in the code -- we'd be trying to set a negative value here -- but it's a bug that we aren't catching currently.

To catch this, capture the return value of the `exec_query` and error if it doesn't equal 1.

(I suspect this error case is what was triggering the bug from yesterday -- there *should* have been a negative balance violation, but because of the delete-0-rows trigger and this update not updating anything, you weren't getting an error out of it).